### PR TITLE
Beta.1 Release Notes

### PR DESCRIPTION
**Pull request by huntc**
... including a doc for external service discovery.

### CODE REVIEW

File: src/main/play-doc/general/ReleaseNotes.md [hunk 1]
```diff
@@ -18,6 +19,16 @@ The new licensing mechanism permits a single ConductR Agent to be used absolutel
 
 Please visit the [migration guide](MigrationGuide#Production_Suite_Licensing) for additional instructions.
 
+### External Service Registration
+
+Since 2.0, ConductR has supported a DNS SRV record fallback mechanism for service discovery i.e. if a service cannot be discovered as being hosted by ConductR, then DNS SRV lookups can be performed in order to resolve the service's location. This was disabled for standalone mode, but enabled for DC/OS installation so that Elasticsearch lookups can utilize the DC/OS hosted service.
+
+As of 2.1, ConductR replaces the 2.0 mechanism with something considerably more flexible. Any service name may be registered via ConductR's configuration to exist as an external service, and one of: an explicit IP and port; DNS and port; or DNS SRV host may be supplied. Furthermore any number of these can be mapped to a service name and also resolved with user credentials (http basic auth at this point), and a root path.
```
Comment (markusjura) [inline src/main/play-doc/general/ReleaseNotes.md]: `and one of: an explicit IP and port; DNS and port; or DNS SRV host may be supplied`

Can we create a separate bullet point for each item.

Comment (markusjura) [reply src/main/play-doc/general/ReleaseNotes.md]: Also, it is possible to specify a set of URIs and not only one? The documentation does not point that out clearly enough.

File: src/main/play-doc/api/BundleAPI.md [hunk 2]
```diff
@@ -61,6 +61,141 @@ Returns a list of host and port of the running service given a service name.
...
+  {
+    "protocol":"http",
+    "ipv4Host":"10.11.23.22",
+    "port":7002
+  }
+]
+```
+
+Where `10.11.23.22:7001` and `10.11.23.22:7002` are the host and host port of the service which is currently running, and the protocol
+indicates the protocol to be used for connectivity. Additional fields include `authorization` which presently holds just `basic` reflecting HTTP basic authorization, along with correspondong `userinfo`, and `path` which provides an root path to prefix any requests with in the case of HTTP. These fields are optional. Another is `ipv6` which is the corresponding IPv6 address, noting that the `ipv4Host` may be optional where `ipv6` is provided.
```
Comment (markusjura) [inline src/main/play-doc/api/BundleAPI.md]: A table would be nice to list all required and optional response fields.

**Review by markusjura — CHANGES_REQUESTED** [review]
Overall, looking good. Only some minor comments.

File: src/main/play-doc/general/ReleaseNotes.md (hunk 1 above)
Comment (huntc) [reply src/main/play-doc/general/ReleaseNotes.md]: Done

File: src/main/play-doc/api/BundleAPI.md (hunk 2 above)
Comment (huntc) [reply src/main/play-doc/api/BundleAPI.md]: Done

File: src/main/play-doc/api/BundleAPI.md [hunk 3]
```diff
@@ -61,6 +61,150 @@ Returns a list of host and port of the running service given a service name.
...
+
+Field            | Description
+-----------------|------------
+protocol         | The protocol to be used for connectivity.
+authorization    | This presently holds just `basic` reflecting HTTP basic authorization. Optional.
+userInfo         | HTTP user info supplied as a username and password delimited with a `:`. Optional but mandatory if `authorization` reflects `basic`.
+ipv4Host         | The IPv4 representation of the address. Optional.
+ipv6Host         | The IPv6 representation of the address. Optional (generally favor IPv6 addresses if there are both).
+port             | The port to connect to.
+path             | The root context path to prepend to any other path in the case of HTTP.
```
Comment (markusjura) [inline src/main/play-doc/api/BundleAPI.md]: The `path` is optional, right? If so we should mention that.

File: src/main/play-doc/api/BundleAPI.md [hunk 4]
```diff
@@ -61,6 +61,150 @@ Returns a list of host and port of the running service given a service name.
...
+    "protocol":"http",
+    "ipv4Host":"10.11.23.22",
+    "port":7002
+  }
+]
+```
+
+Where `10.11.23.22:7001` and `10.11.23.22:7002` are the host and host port of the service which is currently running. The following table shows the complete list of fields available to a reply:
+
+Field            | Description
```
Comment (markusjura) [inline src/main/play-doc/api/BundleAPI.md:94]: Can we order the fields either alphabetically or sorted by required and optional? I'd add the information to a field if it is required. So far we only describe that a given field is optional.

File: src/main/play-doc/api/BundleAPI.md [hunk 5]
```diff
@@ -98,7 +242,7 @@ Either the service has not been started, or the service requested cannot be foun
 
 Other status codes should also be treated as a failure.
 
-## Receive hosts events
+## Receive hosts events (deprecated)
```
Comment (markusjura) [inline src/main/play-doc/api/BundleAPI.md]: This API has been deprecated. We should describe what to use instead.

File: src/main/play-doc/api/BundleAPI.md [hunk 6]
```diff
@@ -135,7 +279,7 @@ Heartbeat events are also emitted so that the connection is kept alive through h
 
 Other status codes should also be treated as a failure.
 
-## Find host service
+## Find host service (deprecated)
```
Comment (markusjura) [inline src/main/play-doc/api/BundleAPI.md]: This API has been deprecated. We should describe what to use instead.

File: src/main/play-doc/general/ReleaseNotes.md [hunk 7]
```diff
@@ -18,6 +19,22 @@ The new licensing mechanism permits a single ConductR Agent to be used absolutel
 
 Please visit the [migration guide](MigrationGuide#Production_Suite_Licensing) for additional instructions.
 
+### External Service Registration
+
+Since 2.0, ConductR has supported a DNS SRV record fallback mechanism for service discovery i.e. if a service cannot be discovered as being hosted by ConductR, then DNS SRV lookups can be performed in order to resolve the service's location. This was disabled for standalone mode, but enabled for DC/OS installation so that Elasticsearch lookups can utilize the DC/OS hosted service.
+
+As of 2.1, ConductR replaces the 2.0 mechanism with something considerably more flexible. Any service name may be registered via ConductR's configuration to exist as an external service given one of the following mappings:
+
+* an explicit IP and port;
```
Comment (markusjura) [inline src/main/play-doc/general/ReleaseNotes.md]: Semicolons, fullstops and or's should not be used in bullet points. Also, the first letter should be upper case. What about:

* IP address and port
* DNS address and port
* DNS SRV host

**Review by markusjura — CHANGES_REQUESTED** [review]
Requesting some styling changes.

File: src/main/play-doc/api/BundleAPI.md (hunk 4 above)
Comment (huntc) [reply src/main/play-doc/api/BundleAPI.md:94]: yes

File: src/main/play-doc/api/BundleAPI.md (hunk 3 above)
Comment (huntc) [reply src/main/play-doc/api/BundleAPI.md]: yes - thanks

File: src/main/play-doc/api/BundleAPI.md (hunk 5 above)
Comment (huntc) [reply src/main/play-doc/api/BundleAPI.md]: ok

Comment (huntc) [reply src/main/play-doc/api/BundleAPI.md]: done

File: src/main/play-doc/api/BundleAPI.md (hunk 6 above)
Comment (huntc) [reply src/main/play-doc/api/BundleAPI.md]: done

File: src/main/play-doc/general/ReleaseNotes.md (hunk 7 above)
Comment (huntc) [reply src/main/play-doc/general/ReleaseNotes.md]: ok

File: src/main/play-doc/api/BundleAPI.md [hunk 8]
```diff
@@ -61,6 +61,150 @@ Returns a list of host and port of the running service given a service name.
...
+
+Where `10.11.23.22:7001` and `10.11.23.22:7002` are the host and host port of the service which is currently running. The following table shows the complete list of fields available to a reply:
+
+Field            | Description
+-----------------|------------
+authorization    | This presently holds just `basic` reflecting HTTP basic authorization. Optional.
+ipv4Host         | The IPv4 representation of the address. Optional.
+ipv6Host         | The IPv6 representation of the address. Optional (generally favor IPv6 addresses if there are both).
+path             | The root context path to prepend to any other path in the case of HTTP. Optional.
+port             | The port to connect to.
```
Comment (markusjura) [inline src/main/play-doc/api/BundleAPI.md:100]: We should document that port and protocol is a required field.

```
port             | The port to connect to. Required.
protocol         | The protocol to be used for connectivity. Required.
```

File: src/main/play-doc/api/BundleAPI.md [hunk 9]
```diff
@@ -98,7 +242,7 @@ Either the service has not been started, or the service requested cannot be foun
 
 Other status codes should also be treated as a failure.
 
-## Receive hosts events
+## Receive hosts events (deprecated - use the /v2 path instead)
```
Comment (markusjura) [inline src/main/play-doc/api/BundleAPI.md]: Can we do more specific about the replacement and add a link to it, e.g.

## Receive hosts events (deprecated - use the [/v2](#Receive-hosts-events) endpoint instead)

The link consists of `[/v2](#Receive-hosts-events)`

Same for the other deprecated endpoints.

**Review by markusjura — CHANGES_REQUESTED** [review]
- Mark required fields as required
- Links to deprecation replacement

File: src/main/play-doc/api/BundleAPI.md (hunk 8 above)
Comment (huntc) [reply src/main/play-doc/api/BundleAPI.md:100]: Presently consistent with other places.

**Review by markusjura — APPROVED** [review]

**Pull request merged**